### DISCUSSION TITLE
Mod mail/Report

### DISCUSSION THREAD

**Discussion by MichelleBellez** [Zrips/CMI#3536] [Feature Requests]
Make a command that sends a mail to all mods/staff admin.
so if a player sees something they report it to all who have permission to receive it 
mostly so that if no admins are online they can report anything that needs to be reported to the staff. and the staff will see it as a mail next time they log in

player sends - _/mail report <message>_
the message will get sent to everyone with _cmi.reportmail_ permission
and they read it with the normal _/mail read_ that pops up when they login

**Comment by CloudeLecaw** [comment]
Why this is a bad idea? You have 20 stuff members, non is there.. Frist geht logged in and look for the mail, next logs on, and right.. Looks for mail... Next logs on and..? Right looks for the mail... And so on.. What you suggestion would just all staffmembers have trying to care over mailreports doubled and trice. Even if problem is allready solved

What is about note? What is a about warn? These commands do simliar thing s allready. Warn give out a warning and note gives a Note. warning can be spreaded into points by cathegory and execute commands when a player collected an amount of points. With notes when player logs in and he has a Note all players with permission to note will get informed. Then they can list, add, remove his notes by /cmi note playername.
On our sever we created a Warnung aliascommand for staffmembers that first warns the player, and creates a note for the player about the reason.

What you where asking for is a report system that is a bit far away from what mail does. If you realy want a report system feature why not suggesting this?
The basis mostly are
-/Report reason
-/report ticket (teleport/close/info)(reason) 
-Information for members on login of open tickets when logged in
-maybe some optiinal extension to answer and communicate
-optional inform+on login when someone communicated the ticked
-If a ticket is closed send mail to creator

But creating the tickets iside the mailsystem sound like a missued idea of the Mail system

**Comment by MichelleBellez** [comment]
I have a small server with only 4 staff members. So I figured attaching it to the pre-built mail system that notify the player when they log in would be easier than building a new system. 
basically a group mail system so you could notify a pre-determined group of people (in this case a player reporting something to all Staff). 

I do agree I could have worded my suggestion better and a /report command would work well or the issue as well.

**Comment by CloudeLecaw** [comment]
just by the way, i renember a report feature was suggested a few times allready, but maybe they would now see the needing for it :)

just lets wait what Zrips is saying about this. personaly i also would like to see some kind of reportsystem build into cmi as a feature ;)

**Comment by mrfloris** [comment]
Hey Michelle,

I like the idea of a system where the player can get in contact with the staff, even if they are offline. This way when the appropriate staff shows up then someone can pick up this mail (or report) and process it. 

I vote for a mixture between mail, helpop, staffmsg and report system. 

A full fledged report system might not be the ideal way, it can get bloated, and people want special features. It feels like this would be better suited for a standalone plugin dedicated to dealing with the unique way tickets are handled on the system.

A mail system is a nice idea, but as Cloud pointed out, this could mean if one staff member picks up a mail and has handled it, that when others login they also have those mails. Not knowing the progress of the ticket. 

And /staffmsg is nice, but this is for online staff only. And only staff can use it. The same for /helpop, it's for users, but doesn't store anything, and is for online staff only.

A simple system that works like /mail, but is for reports, that informs online staff members there's a new ticket submitted, but when 1 user clicks on it, it's then assigned to that staff member (and others click on it they get the msg 'already assigned', might be a compromise. The details of this is something worth discussing.

I like the idea of finding "1" cmi way of dealing with users who need to leave a msg for staff (and can't for some reason use a forum, mail, or ticket system or discord...)
